src/main/java/seedu/address/model/country/Country.java

```diff
@@ -1,16 +1,69 @@
 package seedu.address.model.country;
 
-public enum Country {
-//    TODO: Add implementation
-//    private Set<Note> noteList;
-//    public Set<Note> getNoteList() {
-//        if (noteList != null) {
-//            return Collections.unmodifiableSet(noteList);
-//        } else {
-//            return new HashSet<>();
-//        }
-//    }
-//    public void setNoteList(Set<Note> noteList) {
-//        this.noteList = noteList;
-//    }
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+import seedu.address.model.note.Note;
+
+/**
+ * A representation of a Country that can be identified by a 2-letter ISO3166 country-code or by its country
+ * name. It contains a list of country notes.
+ */
+public class Country {
+
+    private final String countryName;
+    private final String countryCode;
+    private final ArrayList<Note> countryNotes;
+
+    /**
+     * Initializes a Country by its countryCode.
+     *
+     * @param countryCode The ISO3166 2-letter country code of the country to be initialized.
+     */
+    public Country(String countryCode) {
+        this.countryCode = countryCode;
+        this.countryName = new Locale("", countryCode).getDisplayName();
+        this.countryNotes = new ArrayList<>();
```

Comment:
I feel like we shouldn't initialize countryNotes until it is actually needed. E.g. Only when the user actually adds a country note, or there are notes for this country in the save file. If not its unnecessary overhead, especially if there are many countries.

Not sure how the rest think @tankangliang @rtshkmr  @LeeEnHao

Reply:
I think it's okay to init it for now. We could probably KIV this for further optimisations if it's ever a problem down the road? 
I agree with the case when there will be many countries and this poses a real problem though.

Initialising probably makes for easy testing for now as well.

Reply:
Yea that was the intention to just init countryNotes from the start so there wouldn't be any bugs if someone using Country forgot to check if countryNotes exists. If there is a noticeable lag then probably can optimise in v1.3.

Save file can just store countries which actually have countryNotes in them, so no need to store all countries.